src/data-workspace/use-attribute-option-combo.js

```diff
@@ -1,43 +1,52 @@
 import { useMemo } from 'react'
 import { useContextSelection } from '../context-selection/index.js'
-import { useMetadata } from '../metadata/index.js'
+import { useMetadata, useAttributeMetadata } from '../metadata/index.js'
 import {
-    getCoCByCategoryOptions,
     getDataSetById,
     getCategoryComboById,
+    findAoCByCategoryOptions,
 } from '../metadata/selectors.js'
 
 // TODO: this should probably be handled by useContextSelection-hook
 // should not need this when api support CC and CP instead of cocId
 export const useAttributeOptionCombo = () => {
     const { isLoading, isError, data } = useMetadata()
+    const attributeMetadata = useAttributeMetadata()
     const [{ dataSetId, attributeOptionComboSelection }] = useContextSelection()
 
     const cocId = useMemo(() => {
-        if (!isLoading && !isError && dataSetId) {
+        const hasData = data && attributeMetadata.data
+        if (hasData && dataSetId) {
```

Comment:
Why remove the `!isLoading && !isError && ` part? At some point we'll add a UI so the user can refetch metadata? Or is that irrelevant for this part and we want to show stale data?

Reply:
Basically because it's not needed, we only care about the data. `isLoading` is only true in case of "hard" loading, eg. no cached data, so "first" load. Therefore data is undefined as well. If we want the behaviour you are talking about we would want to use `isRefetching`.